cpp_api/options.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
```

Comment:
I think we should be using header guards rather than "pragma once", in order to match the style of the rest of the code base.